### PR TITLE
CP-18194: Use QMP to change CD media (insert/remove)

### PR DESCRIPTION
**Pull request by fillzero**
Signed-off-by: Liang Dai <liang.dai1@citrix.com>

### CODE REVIEW

**Comment by fillzero** [comment]
This PR depends on https://github.com/xapi-project/ocaml-qmp/pull/9

**Comment by mseri** [comment]
You don't need to add `setup.ml` and `_tags`. They are re-generated when we package xenopsd. We can actually remove some of their content. Please run `oasis setup-clean -replace-sections -remove` then add the setup file changes in a separate commit, this will remove unnecessary boilerplate.

File: xc/device.ml [hunk 1]
```diff
@@ -603,7 +603,21 @@ let qemu_media_change ~xs device _type params =
 		"params",         pathtowrite;
 	] in
 	Xs.transaction xs (fun t -> t.Xst.writev backend_path back_delta);
-	debug "Media changed: params = %s" pathtowrite
+	debug "Media changed: params = %s" pathtowrite;
+
+	try
+		ignore(with_xs (fun xs -> xs.Xs.read (Printf.sprintf "/libxl/%d/dm-version" device.frontend.domid)) = "qemu_xen");
+		let c = Qmp_protocol.connect (Printf.sprintf "/var/run/xen/qmp-libxl-%d" device.frontend.domid) in
```
Comment (mseri) [inline xc/device.ml]: I don't like how we do the discovery right now, but I think this can go in while a more global decision on the design is taken

**Comment by mseri** [comment]
The changes to xs-opam have been merged, but you will need a new release of xs-opam for your feature branch to be able to use the new changes. Ping me on Slack and I can tell you how to do it

**Comment by fillzero** [comment]
Appreciate your help.

**Pull request merged**